src/test/java/com/google/step/snippet/external/W3SchoolsClientTest.java

```diff
@@ -44,19 +45,19 @@ public void jsonCodeCard() {
 
   @Test
   public void partiallyFilledCard() {
-    Card actual = client.search("https://www.w3schools.com/html/html_quiz.asp");
+    Card actual = client.search("https://www.w3schools.com/html/html_quiz.asp", "html quiz");
     assertNull(actual);
   }
 
   @Test
   public void nonexistentLink() {
-    Card actual = client.search("https://www.w3schools.com/css/html.asp");
+    Card actual = client.search("https://www.w3schools.com/css/html.asp", null);
```

Comment:
Let's use `""` instead of `null`. Your code implicitly assumes the query is non-null since it calls `toLowerCase()` on it.